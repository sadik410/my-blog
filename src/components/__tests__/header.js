import * as React from 'react';
import { render, screen } from "@testing-library/react";

import { useStaticQuery } from "gatsby";
import Header from "../header/Header";

const links = [
  { link: "/", title: "Home" },
  { link: "/Resume", title: "Resume" },
  { link: "/Services", title: "Services" },
];

beforeEach(() => {
  useStaticQuery.mockImplementationOnce(() => ({
      site: {
        siteMetadata: {
          header: {
            menus: links,
          },
        },
      },
    })
  );
});

describe("Header", () => {
  it("renders correctly", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
  it("renders a title with my name", () => {
    const { getByText } = render(<Header />);
    expect(getByText((content, element) => {
        return element.tagName.toLowerCase() === 'h1' && element.textContent === 'Sadik Milliti'
    })).toBeInTheDocument();
  });
  it("renders the links in the gatsbyConf", () => {
    const { getByText } = render(<Header />);
    links.forEach(item => {
      const getLink = (content, element) => {
        return element.tagName.toLowerCase() === 'a' && element.textContent === item.title
      }
      expect(getByText(getLink)).toBeInTheDocument();
      expect(getByText(getLink).getAttribute('href')).toEqual(item.link);
    })
  });
});
