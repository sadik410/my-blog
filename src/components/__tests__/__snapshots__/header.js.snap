// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <header
    style="padding-top: 20px; padding-bottom: 20px;"
  >
    <div
      class="header row justify-content-between"
    >
      <h1
        class="name"
      >
        Sadik
        <span
          style="color: rgb(33, 150, 243);"
        >
           Milliti
        </span>
      </h1>
      <nav>
        <ul
          class=""
        >
          <li>
            <a
              class="link"
              href="/"
              state="[object Object]"
            >
              Home
            </a>
          </li>
          <li>
            <a
              class="link"
              href="/Resume"
              state="[object Object]"
            >
              Resume
            </a>
          </li>
          <li>
            <a
              class="link"
              href="/Services"
              state="[object Object]"
            >
              Services
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</div>
`;
